src/parser_urdf.cc

```diff
@@ -3200,6 +3200,35 @@ void CreateVisual(tinyxml2::XMLElement *_elem, urdf::LinkConstSharedPtr _link,
   // set additional data from extensions
   InsertSDFExtensionVisual(sdfVisual, _link->name);
 
+  if (_visual->material)
+  {
+    double color[4];
+    color[0] = _visual->material->color.r;
+    color[1] = _visual->material->color.g;
+    color[2] = _visual->material->color.b;
+    color[3] = _visual->material->color.a;
+    auto materialTag = sdfVisual->FirstChildElement("material");
+    // If the material is not included by an extension then create it
+    if (materialTag == nullptr)
+    {
+      AddKeyValue(sdfVisual, "material", "");
+      materialTag = sdfVisual->FirstChildElement("material");
+    }
+    // If the specular and diffuse are defined by an extension then don't
+    // use the color values
+    if (materialTag->FirstChildElement("diffuse") == nullptr &&
```

Comment:
I recently had a discussion about how material tags are translated to urdf color. In the [sdformat_urdf](https://github.com/ros/sdformat_urdf/blob/ros2/sdformat_urdf/README.md#material-limitations) package, this formula was used `urdf color = 0.4 * sdf ambient + 0.8 * sdf diffuse` and the related discussion is here: https://github.com/ros/sdformat_urdf/pull/1#discussion_r509709647

I wonder if we could do the reverse here. I don't have a good suggestion on how though.

Reply:
Yep, there are 3 equations, each one of the equations have two unknown variables, and there is no relation between them. I neither have a good suggestion. We only know that this variables should live in the interval [0, 1].

Reply:
I'm against adding any ambient color by default. That would cause problems in dark areas where the models would glow. Imagine a SubT-like scenario - you don't want any ambient color there...

Reply:
That would mean `sdf diffuse = 1.25 * urdf color, sdf ambient = 0`

Reply:
@peci1 I don't think adding ambient would cause objects to glow. That would be emissive. Citing http://gazebosim.org/tutorials?tut=color_model:
> The final color of an object depends both on the material and the lights shining on it. Values on lights represent the intensity of light emitted. Values on a material represent the percentage of light an object reflects. The same material may be different colors depending on the light that hits it.

So, if the ambient of the scene is [0, 0, 0], then the ambient component of the objects material won't affect the final color of the object. For SubT, the scene's ambient is [0.1, 0.1, 0.1], so, each of the object's ambient channels would be multiplied by 0.1.

Reply:
Hmm, so the influence of a material's `<ambient>` color depends on the global `<ambient>`? Then it would be okay to set some - if somebody wanted to make a totally dark world, it would be possible by setting global ambient to black.

Reply:
+1 to setting both ambient and diffuse to the URDF color, for a "plain color" look. Specular and emissive feel like "extra" features. Users can always use the extension if they need finer control.

Reply:
https://github.com/osrf/sdformat/pull/526/commits/c58aa5f4bcf5191fc7843fc45005c36a7a09478b

Reply:
Thinking through [this](https://github.com/ros/sdformat_urdf/pull/1#discussion_r514517271) comment a little more, I think we can assign `sdf diffuse = 1.25 * urdf` and `sdf ambient = 1.25 * urdf`. 

My reasoning is that [Rviz](https://github.com/ros-visualization/rviz/blob/8bd56fb928b255e095424a5f1fb3ff8a5274602e/src/rviz/robot/robot_link.cpp#L472-L473) takes the urdf color and sets ` urdf diffuse = urdf` and `urdf ambient = 0.5 * urdf` with an ambient light of `rgba(1 1 1 1)` per [this](https://github.com/ros/sdformat_urdf/pull/1#discussion_r514517271) comment, and diffuse light of `rgba (1 1 1 1)` [as well](https://github.com/ros-visualization/rviz/blob/8bd56fb928b255e095424a5f1fb3ff8a5274602e/src/rviz/visualization_manager.cpp#L175). This means, the final diffuse and ambient colors are set by `urdf diffuse` and `urdf ambient`. In Gazebo, let's assume the default ambient light of `rgba(0.4 0.4 0.4 1)` and diffuse light of `rgba(0.8 0.8 0.8 1)`. The material colors are multiplied by these to obtain the final colors, so to get the same final colors as in Rviz, we should do `sdf ambient = urdf ambient / 0.4` and `sdf diffuse = urdf diffuse / 0.8`. But since  ` urdf diffuse = urdf` and `urdf ambient = 0.5 * urdf` , this results in `sdf ambient = 0.5 * urdf / 0.4` and `sdf diffuse = urdf / 0.8`.

We do need to clamp the values to 1.0 after doing the math because values > 1.0 will result in an error (see #519).